src/emulator/errordisplay.c

```diff
@@ -0,0 +1,669 @@
+/**
+ * @file errordisplay.c
+ *
+ * This file implements the screen that can be viewed during the emulator's initialization process.
+ * In most cases it prints "You will need the Classic Controller."
+ * but an error message can be printed if something goes wrong.
+ *
+ * This file is also responsible for reading the banner title and subtitle from the string table.
+ */
+#include "emulator/errordisplay.h"
+#include "emulator/banner.h"
+#include "emulator/controller.h"
+#include "emulator/frame.h"
+#include "emulator/rom.h"
+#include "emulator/system.h"
+#include "emulator/vc64_RVL.h"
+#include "emulator/xlCoreRVL.h"
+#include "emulator/xlFile.h"
+#include "emulator/xlHeap.h"
+#include "macros.h"
+#include "revolution/demo.h"
+#include "revolution/nand.h"
+#include "revolution/os.h"
+#include "revolution/vi.h"
+
+static s32 fn_80063680(EDString* pEDString);
+static s32 errordisplayReturnToMenu(EDString* pEDString);
+
+static EDStringInfo sStringBase[] = {
+    {SID_ERROR_INS_SPACE, 0, NULL, 0x00000000, 0x00000000},
+    {SID_ERROR_CHOICE_PRESS_A_TO_RETURN_TO_MENU, 0, NULL, 0x00000000, 0x00000000},
+    {SID_ERROR_INS_INNODE, 0, NULL, 0x00000000, 0x00000000},
+    {SID_ERROR_SYS_CORRUPT, 0, NULL, 0x00000000, 0x00000000},
+    {SID_ERROR_DATA_CORRUPT, 0, NULL, 0x00000000, 0x00000000},
+    {SID_ERROR_MAX_BLOCKS, 0, NULL, 0x00000000, 0x00000000},
+    {SID_ERROR_MAX_FILES, 0, NULL, 0x00000000, 0x00000000},
+    {SID_ERROR_NO_CONTROLLER, 0, NULL, 0x00000000, 0x00000000},
+    {SID_ERROR_NEED_CLASSIC, 0, NULL, 0x00000000, 0x00000000},
+    {SID_ERROR_REMOTE_BATTERY, 0, NULL, 0x00000000, 0x00000000},
+    {SID_ERROR_REMOTE_COMMUNICATION, 0, NULL, 0x00000000, 0x00000000},
+    {SID_ERROR_BLANK, 0, NULL, 0x00000000, 0x00000000},
+    {SID_NONE, 0, NULL, 0x00000000, 0x00000000},
+};
+
+ErrorDisplay sStringDraw[] = {
+    {
+        {&sStringBase[SI_ERROR_INS_SPACE], FLAG_COLOR_WHITE, 0, 0},
+        {{
+            &sStringBase[SI_ERROR_CHOICE_PRESS_A_TO_RETURN_TO_MENU],
+            FLAG_COLOR_WHITE,
+            0,
+            0,
+            (ErrorCallback)errordisplayReturnToMenu,
+        }},
```

Comment:
the formatting here seems inconsistent (due to whether the inner struct here has a trailing comma)